src/components/BookItem.js

```diff
@@ -1,40 +1,48 @@
 import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/booksSlice';
 
-const BookItem = ({ itemProp, delBook }) => (
-  <li className="book-container">
-    <div className="book-info">
-      <span className="genre">Science Fiction</span>
-      <h3 className="title">{itemProp.title}</h3>
-      <span className="author">{itemProp.author}</span>
-      <div className="edit-info">
-        <button type="button">Comments</button>
-        <hr />
-        <button type="button" onClick={() => delBook(itemProp.id)}>
-          Remove
-        </button>
-        <hr />
-        <button type="button">Edit</button>
+const BookItem = ({ itemProp }) => {
+  const dispatch = useDispatch();
+  return (
+    <li className="book-container">
+      <div className="book-info">
+        <span className="genre">{itemProp.category}</span>
+        <h3 className="title">{itemProp.title}</h3>
+        <span className="author">{itemProp.author}</span>
+        <div className="edit-info">
+          <button type="button">Comments</button>
+          <hr />
+          <button
+            type="button"
+            onClick={() => dispatch(removeBook(itemProp, itemProp.item_id))}
+          >
```

Comment:
- [ ] Please kindly work more on the remove functionality, when I tried to remove a book from the list of newly added books; all the newly added books were removed (though this was not so for the books in the initial state). I guess the same key is assigned for all the childrem. Kindly check your browser console and this [video](https://www.loom.com/share/4c80619613a34f5da386a58e90836efe) for more clarification. 👍

Reply:
Hi @OyePriscilla I really Really really do appreciate your video, thanks, The error was fixed now!

Reply:
Hi @OyePriscilla I really Really really do appreciate your video, thanks, The error was fixed now!